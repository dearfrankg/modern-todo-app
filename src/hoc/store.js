import React, { Component } from "react";
import { connect } from "react-redux";

export default function(ComposedComponent, reducers) {
  class decoreWithStore extends Component {
    componentWillMount() {
      // eslint-disable-next-line
      // console.log('Will Mount HOC', reducers);
    }

    render() {
      return <ComposedComponent {...this.props} />;
    }
  }

  function mapStateToProps(state) {
    // eslint-disable-next-line
    if (reducers.length === 0) console.error("There' no reducer here, fix it!", reducers);

    const reducerMap = {};

    reducers.forEach(reducer => {
      reducerMap[reducer] = state[reducer];
    });

    return reducerMap;
  }

  return connect(mapStateToProps)(decoreWithStore);
}
