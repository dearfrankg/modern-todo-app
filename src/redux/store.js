import { createStore, combineReducers } from 'redux';
import { routerReducer } from 'react-router-redux';

import rootReducer from 'redux/reducers';
import { loadState, saveState } from 'redux/localStorage';

const persistedState = loadState();

const store = createStore(
  combineReducers({
    todos: rootReducer,
    router: routerReducer,
  }),
  persistedState,
);

store.subscribe(() => {
  saveState({
    todos: store.getState().todos,
  });
});

export default store;
