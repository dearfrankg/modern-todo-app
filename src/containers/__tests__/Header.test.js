import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';

import store from 'redux/store';
import Container, { HeaderContainer, mapDispatchToProps } from '../Header';

describe('Header Container:', () => {
  it('should render correctly with default props', () => {
    expect(
      shallow(
        <Provider store={store}>
          <Container />
        </Provider>,
      ),
    ).toMatchSnapshot();
  });

  it('should call onChange when input changed', () => {
    const wrapper = mount(<HeaderContainer addTodo={() => {}} />);

    wrapper.find('input').simulate('change', { target: { value: 'My new value' } });
    expect(wrapper.state().value).toBe('My new value');
  });

  it('should call onSubmit when pressing enter on input', () => {
    const wrapper = mount(<HeaderContainer addTodo={() => {}} />);

    wrapper.find('input').simulate('keyDown', { key: 'Enter', keyCode: 13, which: 13 });
    expect(wrapper.state().value).toBe('');
  });

  it('should call onSubmit and update state when pressing esc on input', () => {
    const wrapper = mount(<HeaderContainer addTodo={() => {}} />);

    wrapper.find('input').simulate('keyDown', { key: 'Esc', keyCode: 27, which: 27 });
    expect(wrapper.state().value).toBe('');
  });

  it('should call onSubmit and update state when pressing esc on input', () => {
    const wrapper = mount(<HeaderContainer addTodo={() => {}} />);

    wrapper
      .find('input')
      .props()
      .onKeyDown({
        target: { value: 'teste' },
        which: 13,
      });

    expect(wrapper.state().value).toBe('');
  });

  describe('should call actions', () => {
    const dispatchSpy = jest.fn();
    const { addTodo } = mapDispatchToProps(dispatchSpy);

    it('should call addTodo', () => {
      addTodo();
      const spyLastCall = dispatchSpy.mock.calls[0][0];
      expect(spyLastCall.type).toEqual(dispatchSpy.mock.calls[0][0].type);
    });
  });
});
