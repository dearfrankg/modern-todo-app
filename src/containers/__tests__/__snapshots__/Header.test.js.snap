// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Container: should render correctly with default props 1`] = `<Connect(HeaderContainer) />`;
